source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,369 @@
+   See the `git-format-patch Documentation`_ for detailed information 
+   on using :command:`git format-patch`
+
+   .. code-block:: bash
+      
+      git format-patch -<n>
+
+
+#. Copy the patch files from the patches directory in the linux 
+   source tree to the RPM build directory.
+
+   .. code-block:: bash
+      
+      cp patches/*.patch ~/clearlinux/packages/linux/
+
+
+#. Navigate back to the RPM build directory.
+
+   .. code-block:: bash
+      
+      cd ~/clearlinux/packages/linux/
+
+
+#. Open the Linux kernel package RPM SPEC file in an editor.
+
+   .. code-block:: bash
+
+      $EDITOR linux.spec
+
+
+#. Locate the section of the SPEC file that contains existing patch 
+   variable definitions and append your patch file name. Ensure the 
+   patch number does not collide with an existing patch.
+
+   .. code-block:: bash
+
+      #
+      # Small Clear Linux Tweaks
+      #
+      Patch0501: 0501-zero-extra-registers.patch
+      Patch0502: 0502-locking-rwsem-spin-faster.patch
+
+      #Serie1.name WireGuard
+      #Serie1.git  https://git.zx2c4.com/WireGuard
+      #Serie1.tag  00bf4f8c8c0ec006633a48fd9ee746b30bb9df17
+      Patch1001: 1001-WireGuard-fast-modern-secure-kernel-VPN-tunnel.patch
+      #Serie1.end
+
+      Patch2001: 2001-my-patch-for-driver-A.patch
+
+
+#. Locate the section of the SPEC file further down that contains 
+   patch application and append your patch file number used in the step above.
+
+   .. code-block:: bash
+
+      #
+      # Small tweaks
+      #
+      %patch0501 -p1
+      %patch0502 -p1
+
+      #Serie1.patch.start
+      %patch1001 -p1
+      #Serie1.patch.end
+
+      %patch2001 -p1
+
+
+#. Commit and save the changes to the RPM SPEC file.
+
+
+
+Build and install the kernel
+****************************
+After changes have been made to the kernel SPEC file and config file, 
+the kernel is ready to be compiled and packaged into a RPM.
+
+#. Start the compilation process by issuing the :command:`make build` command. 
+   This process is typically resource intensive and will take a while.
+
+   .. code-block:: bash
+      
+      make build
+
+
+#. The result there will be multiple :file:`.rpm` files in the :file:`rpms` 
+   directory as output. 
+
+   .. code-block:: bash
+      
+      ls rpms/
+
+   The kernel RPM will be named :file:`linux<name>-<version><release>.x86_64.rpm`
+
+
+#. The kernel RPM file can be input to the `mixer tool`_ to create a custom 
+   bundle and mix of |CL|.
+
+   
+
+Alternatively, the kernel RPM bundle can be installed manually on a local 
+machine for testing. This approach works well for individual development or 
+testing. For a more scalable and customizable approach, consider using the 
+`mixer tool`_ to provide a custom kernel with updates.
+
+1. Install the kernel RPM onto the local system with the :command:`rpm` command.
+
+   .. code-block:: bash
+
+      sudo rpm -ivh linux-custom.<version>.<release>.x86_64.rpm
```

Comment:
Might be better to provide the rpm2cpio workflow instead; this may otherwise confuse people into thinking we're RPM-based.

Reply:
Agreed and done.